src/sequence.lisp

```diff
@@ -32,6 +32,25 @@
     (t
      (not-seq-error seq))))
 
+(defun string-reverse (string)
+  (let* ((length (length string))
+         (new-string (make-string length)))
+    (dotimes (index length new-string)
+      (setf (aref new-string index) (aref string (- length (1+ index)))))))
+
+(defun vector-reverse (vector)
```

Comment:
vector-reverse and string-reverse can be unified if you use `(make-array size :element-type (array-element-type array))`.

But I am happy with this as it is too if you prefer to leave it. It's up to you.